test/ibft.js

```diff
@@ -221,61 +256,171 @@ contract('Ibft.js', (accounts) => {
+
+        let header = await ibft.m_blockheaders(TESTCHAINID, block.hash);
 
-        validationReceipt = await ibft.SubmitBlock(TESTCHAINID, rlpHeader.unsigned, rlpHeader.signed, rlpHeader.seal, storage.address);
-        event = validationReceipt.receipt.rawLogs.some(l => { return l.topics[0] == '0x' + sha3("AddedBlock()") });
+        // Separate fetched header info
+        parentHash = header[2];
+
+        // Assert that block was persisted correctly
+        assert.equal(parentHash, block.parentHash);
+
+        chainHead = await ibft.m_chainHeads(TESTCHAINID);
+        assert.equal(chainHead, block.hash);
+
+        // Check new validators that are changed in this block
+        let chainValidatorsRoot = await ibft.getValidatorsRoot.call(TESTCHAINID);
+        assert.equal(chainValidatorsRoot, expectedHashValidators5)
+
+      })
+
+      it('Successful Submit block - 8 validators', async () => {
+        // get block and validators from samples
+        block = testBlocks.validators_8.block
+        validators = testBlocks.validators_8.validators
+        expectedHash = bufferToHex(keccak256(Web3EthAbi.encodeParameter("address[]", validators.map(x => x.toLowerCase()).sort())));
+
+        // start
+        await ibft.RegisterChain(TESTCHAINID, validators, block.parentHash, storage.address);
+
+        let chainHead = await ibft.m_chainHeads(TESTCHAINID);
+        assert.equal(chainHead, block.parentHash);
+
+        rlpHeader = encoder.encodeIbftHeader(block);
+
+        // Submit block should succeed
+        let start = Date.now()
+        txToBenchmark = await ibft.SubmitBlock(TESTCHAINID, rlpHeader.unsigned, rlpHeader.signed, rlpHeader.seal, storage.address, validators);
+        duration = Number( (Date.now() - start) / 1000 ).toFixed(3)
+        
+        console.log("\tGas used to submit block with 8 validators= " + txToBenchmark.receipt.gasUsed.toString() + " gas");
+
+        let event = txToBenchmark.receipt.rawLogs.some(l => { return l.topics[0] == '0x' + sha3("AddedBlock()") });
         assert.ok(event, "Stored event not emitted");
 
-        const submittedEvent = validationReceipt.logs.find(l => { return l.event == 'BlockSubmitted' });
+        const submittedEvent = txToBenchmark.logs.find(l => { return l.event == 'BlockSubmitted' });
         assert.equal(Web3Utils.sha3(rlpHeader.signed), submittedEvent.args.blockHash);
 
         let addedBlockHash = await ibft.m_chainHeads.call(TESTCHAINID);
-        assert.equal(addedBlockHash, block_add.hash);
+        assert.equal(addedBlockHash, block.hash);
 
-        let header = await ibft.m_blockheaders(TESTCHAINID, block_add.hash);
+        let header = await ibft.m_blockheaders(TESTCHAINID, block.hash);
 
         // Separate fetched header info
         parentHash = header[2];
 
         // Assert that block was persisted correctly
-        assert.equal(parentHash, block_add.parentHash);
+        assert.equal(parentHash, block.parentHash);
 
-        // Check new validators
```

Comment:
This is an important bit of logic that probably needs to be replicated with your new block samples. The logic might be the same but its good if we can prove that validator set changes for added or removed validators. What do you think?